frontend/components/auth/auth-page-wrapper.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import React from "react";
+import { ReactNode } from "react";
+import { useAuth } from "@/app/auth/auth-context";
+import { Button } from "@/components/ui/button";
+import { useRouter } from "next/navigation";
+
+type AuthCheck = (user: { isAdmin: boolean } | undefined | null) => boolean;
+
+interface AuthPageWrapperProps extends React.HTMLProps<HTMLDivElement> {
+  children: ReactNode;
+
+  // User access rules
+  authCheck?: AuthCheck; // Custom predicate which is true when user is to be granted access
+  requireAdmin?: boolean;
+  requireLoggedIn?: boolean;
+}
+
+const AuthPageWrapper: React.FC<AuthPageWrapperProps> = ({
+  children,
+  ...props
+}) => {
+  const auth = useAuth();
+  const router = useRouter();
+
+  const finalAuthCheck = (
```

Comment:
do you think naming finalAuthCheck is weird in this component's context cause it's the only authCheck? maybe it can just be `authCheck`

Reply:
Yep, thanks for the suggestions